include/zephyr/sensing/sensing_datatypes.h

```diff
@@ -0,0 +1,341 @@
+
+#include <stdint.h>
+
+/**
+ * @brief Data Types
+ * @addtogroup sensing_datatypes
+ * @{
+ */
+
+/**
+ * @struct sensing_sensor_value_header
+ * @brief sensor value header
+ *
+ * Each sensor value data structure should have this header
+ *
+ * Here use 'base_timestamp' (uint64_t) and 'timestampe_delta' (uint32_t) to
+ * save memory usage in batching mode.
+ *
+ * The 'base_timestamp' is for readings[0], the 'timestamp_delta' is relation
+ * to the previous 'readings'. So,
+ * timestamp of readings[0] is
+ *	header.base_timestamp + readings[0].timestamp_delta.
+ * timestamp of readings[1] is
+ *	timestamp of readings[0] + readings[1].timestamp_delta.
+ *
+ * Since timestamp unit is micro seconds, the max 'timestamp_delta' (uint32_t)
+ * is 4295 seconds.
+ *
+ * If a sensor has batched data where two consecutive readings differ by
+ * more than 4295 seconds, the sensor subsystem core will split them
+ * across multiple instances of the readings structure, and send multiple
+ * events.
+ *
+ * This concept is borrowed from CHRE:
+ * https://cs.android.com/android/platform/superproject/+/master:\
+ * system/chre/chre_api/include/chre_api/chre/sensor_types.h
+ */
+struct sensing_sensor_value_header {
+	/** base timestamp of this data readings, unit is micro seconds */
+	uint64_t base_timestamp;
+	/** count of this data readings */
+	uint16_t reading_count;
+};
+
+/**
+ * @brief Sensor value data structure types based on common data types.
+ * Suitable for common sensors, such as IMU, Light sensors and orientation sensors.
+ *
+ * Sensor subsystem will use scaled fixed point data structure for all sensor values,
+ * aligned the HID spec, using the format v*10^x to present the decimal value,
+ * where the v is integer number, either int8/uint8, int16/uint6, or int32/uint32, depends on
+ * required sensor data precision.
+ *
+ * The scale unit exponent x is int8 type int8 type with encoding meanings (page 68 of
+ * HID spec https://usb.org/sites/default/files/hutrr39b_0.pdf):
+ *
+ * Unit Exponent argument       Power of Ten (Scientific Notation)
+ * 0x00			       1 * 10E0
+ * 0x01			       1 * 10E1
+ * 0x02			       1 * 10E2
+ * 0x03			       1 * 10E3
+ * 0x04			       1 * 10E4
+ * 0x05			       1 * 10E5
+ * 0x06			       1 * 10E6
+ * 0x07			       1 * 10E7
+ * 0x08			       1 * 10E-8
+ * 0x09			       1 * 10E-7
+ * 0x0A			       1 * 10E-6
+ * 0x0B			       1 * 10E-5
+ * 0x0C			       1 * 10E-4
+ * 0x0D			       1 * 10E-3
+ * 0x0E			       1 * 10E-2
+ * 0x0F			       1 * 10E-1
+ *
+ * So, we can have below data present ranges:
+ *
+ * int8 type v:    [-128, 127] *10^[-8, 7]
+ * uint8 type v:   [0,	255] * 10^[-8, 7]
+ * int16 type v:   [-32768, 32767] * 10^[-8, 7]
+ * uint16 type v:  [0,	65535] * 10^[-8, 7]
+ * int32 type v:   [-2147483648,  2147483647] * 10^[-8, 7]
+ * uint32 type v:  [0,	4294967295] * 10^[-8, 7]
+ * int64 type v:   [-9223372036854775808,  9223372036854775807] * 10^[-8, 7]
+ * uint64 type v:  [0,	18446744073709551615] * 10^[-8, 7]
```

Comment:
Also discarded during the rename, please don't add new fixed point formats. Should use zDSP's Q format

Reply:
Note that you can only have an unbounded array as the last member of a C structure.  If you want the timestamp deltas managed as a separate array, you would need to pass these to the callback as separate parameters.  This could look like:

```
typedef void (*sensing_data_event_t)(
		int handle,
		struct sensing_sensor_value_header *header,
		void *data,
		uint32_t *timestamp_deltas,
		size_t size)
```

However, this splits the sensor readings into 3 segments instead of being contained in a single memory chunk.

Reply:
@yperess - The Q format was agree on by the Zephyr community.
@huhebo - Would like more details on how to the conversion from sensor data to Q format.  How do sensor algorithms interpret this data?
Ke A Han - what about simple sensors?  @yperess - use of Q format should be used only by sensors that need floating point support.  The data can be integer type for simple sensors, and Q format when needed.  Each sensor type will use the same type (integer vs Q format).
@huhebo will review the DSP library and Q format documentation.

Reply:
After review the details of `zdsp` library (thanks @teburd help me to understand more details), now I understand the `zdsp` is agnostic to architecture with pure C code implementation on no DSP instruction supported architectures, and if the architecture supports DSP instruction can implement the optimized version.  

For architectures like Intel ISH which have no DSP instruction but has FPU, the best solution is to convert `Q Format` to `float` and use `float` based algorithms instead of calling `zdsp` `Q Format` based math lib (pure C code implementation in this case), after algorithms computing, then convert back to `Q Format` and report to `sensing subsystem` for other consumers.  

Thanks @yperess 's so quick PR of conversion utilities for `Q Format` and `float` :  https://github.com/zephyrproject-rtos/zephyr/pull/58100

And according https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1198403600, will keep per reading `timestamp_delta`. And per sensor's nature, will define sensor value data structure base on sensor type, for example, IMU sensors can use `sensing_sensor_value_3d_q31`,  and light sensor, step counter sensor will still use 
`sensing_sensor_value_uint32` etc. 

Will change `sensing_sensor_value_3d_int32` to `sensing_sensor_value_3d_q31` :

```c
struct sensing_sensor_value_3d_q31 {
	struct sensing_sensor_value_header header;
        int8_t shift_x;
        int8_t shift_y; 
        int8_t shift_z;          
	struct {
                uint32_t timestamp_delta;
		union {
			q31_t v[3];
			struct {
				q31_t x;
				q31_t y;
				q31_t z;
			};
		};
	} readings[1];
};
```
The `shift_x/shift_y/shift_z` is the scaling shift bits for full range of the sensor data, for example for a accelerometer sensor, we can use the shift value of `6` to make the range of [-64g, 64g) of the `Q Format` presentation. 

Will update the PR (including doc), and remove not implemented data structure at current, new data structure (per sensor type) will be added in following PRs with new sensor type implemented.